xbmc/platform/linux/LogindUtils.h

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "platform/linux/DBusUtil.h"
+
+class CLogindUtils
```

Comment:
The split between static and non-static member functions in this class is a bit strange IMO

Reply:
Yea so this is the part that I wanted some help with.

Mainly that the Open and Close methods need to be static because of the static function in `CLibInputHelper`. Now I could use the `user_data` parameter there to pass in a class, but I'm not sure if that's a better solution?

I need to have a way to let people building without logind stuff still have an open and close method.

Ideas?

Reply:
Using `user_data` looks OK to me